### PR TITLE
Dwell & Travel time tables

### PR DESCRIPTION
**Pull request by rymarczy**
This commit adds schema and logic for dwell and travel times

These two tables are grouped together because the logic behind them is the same, the only difference being dwell times deals with stopped events and travel times deals with moving events

Asana Task: https://app.asana.com/0/1202841571879768/1202672148025254

### CODE REVIEW

File: performance_manager/lib/l2_dwell_travel_times.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+)
+
+
+def dwell_times(db_manager: DatabaseManager) -> None:
+    """
+    perform update on dwell times table
+
+    executed as delete and insert
+    """
+    process_logger = ProcessLogger("update l2 dwell times")
```
Comment (mzappitello) [inline performance_manager/lib/l2_dwell_travel_times.py:19]: i've been using snake case on the process names to make sure splunk can index them correctly.

File: performance_manager/lib/postgres_schema.py [hunk 2]
```diff
@@ -201,3 +201,39 @@ class TempFullTripEvents(SqlBase):  # pylint: disable=too-few-public-methods
...
+
+
+class TravelTimes(SqlBase):  # pylint: disable=too-few-public-methods
+    """Level 2 Table for Travel Times"""
+
+    __tablename__ = "travelTimes"
+
+    pk_id = sa.Column(
+        sa.Integer,
+        sa.ForeignKey("eventsVehiclePositions.pk_id", ondelete="CASCADE"),
```
Comment (mzappitello) [inline performance_manager/lib/postgres_schema.py:213]: maybe a comment on which event this is going to be pointing to?

File: performance_manager/lib/postgres_schema.py [hunk 3]
```diff
@@ -201,3 +201,39 @@ class TempFullTripEvents(SqlBase):  # pylint: disable=too-few-public-methods
...
+
+
+class DwellTimes(SqlBase):  # pylint: disable=too-few-public-methods
+    """Level 2 Table for Dwell Times"""
+
+    __tablename__ = "dwellTimes"
+
+    pk_id = sa.Column(
+        sa.Integer,
+        sa.ForeignKey("eventsVehiclePositions.pk_id", ondelete="CASCADE"),
```
Comment (mzappitello) [inline performance_manager/lib/postgres_schema.py:231]: same request for comment.

File: performance_manager/lib/l2_dwell_travel_times.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+            VehiclePositionEvents.timestamp_end
+            - VehiclePositionEvents.timestamp_start
+        ).label("dwell_time_seconds"),
+    ).join(
+        FullTripEvents,
+        VehiclePositionEvents.pk_id == FullTripEvents.fk_vp_stopped_event,
+    )
+
+    # pull last created_on timestamp from dwell time table
+    max_create_travel_query = sa.select(sa.func.max(DwellTimes.created_on))
```
Comment (mzappitello) [inline performance_manager/lib/l2_dwell_travel_times.py:36]: how does this scale as `DwellTimes` blows up?

**Comment by mzappitello** [comment]
a few little things but LGTM.

File: performance_manager/lib/l2_dwell_travel_times.py (hunk 4 above)
Comment (rymarczy) [reply performance_manager/lib/l2_dwell_travel_times.py:36]: All of these selects are based on timestamp fields which should be very well optimized in postgres. So I don't think we'll have any issues with the scaling of these operations.

File: performance_manager/lib/l2_dwell_travel_times.py (hunk 1 above)
Comment (rymarczy) [reply performance_manager/lib/l2_dwell_travel_times.py:19]: incorporated

File: performance_manager/lib/postgres_schema.py (hunk 2 above)
Comment (rymarczy) [reply performance_manager/lib/postgres_schema.py:213]: I'm going to change the name of this column to indicate that it's a foreign key `fk_travel_time_id` comment also added to clarify what it's pointing to

File: performance_manager/lib/postgres_schema.py (hunk 3 above)
Comment (rymarczy) [reply performance_manager/lib/postgres_schema.py:231]: I'm going to change the name of this column to indicate that it's a foreign key `fk_dwell_time_id` comment also added to clarify what it's pointing to

**Comment by mzappitello** [comment]
LGTM, squash and merge pls.

**Pull request merged**